vault/audit.go

```diff
@@ -414,6 +420,7 @@ func (c *Core) setupAudits(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	c.auditBroker = broker
```

Comment:
The assignment of the core field should be carried out as soon as we're sure there wasn't an error creating it. This was previously incorrectly done at the end of the func.